test/1.4/pragmas/COVERITY.dml

```diff
@@ -0,0 +1,58 @@
+/*
+  © 2024 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+
+dml 1.4;
+
+device test;
+
+/// COMPILE-ONLY
+/// DMLC-FLAG --coverity
+
+// This test is limited by the fact it relies on test annotations and line
+// directives from generated C. Notably, it has no means of properly testing
+// `--coverity` and the COVERITY pragma in the case of `--noline`.
+
+session struct { const int x; } s = {4};
+// SUPPRESSIONS-ABOVE [store_writes_const_field : FALSE]
+
+template t {
+    method m() {
+        /*% COVERITY foo %*/
+        if (s.x == 4) { after 1 s: m(); }
+        // SUPPRESSIONS-ABOVE [foo]
+    }
+}
+
+is t;
+
+group g is t;
+
+method init() {
+    assert true; // a sanity test that this gets no suppressions
+    // SUPPRESSIONS-ABOVE
+
+    /*% COVERITY irrelevant %*/
+
+    /*% COVERITY foo %*/
+    /*% COVERITY bar %*/ /*% COVERITY baz FALSE %*/
+    assert true;
+    // SUPPRESSIONS-ABOVE [foo] [bar] [baz:FALSE]
```

Comment:
Not sure if this should be `-LAX` or not (we happen to know that `assert` gives a single statement so they are equivalent). Which makes me ask:
```
/*% COVERITY foo %*/
assert true; assert true;
```
... will SUPPRESSIONS-ABOVE hold there? I don't think any other test answers that question.

Reply:
Yes, `SUPPRESSIONS-ABOVE` will hold here, and that **does** get tested via line 23 (many statements/expressions on one line, some of which cause many C statements to be generated.)

Reply:
It should **not** be `-LAX`. One of the guarantees of the `COVERITY` pragma is that, if specified, the annotation for it gets generated for _every_ C line generated from the pragmad DML line. The reason I use `-LAX` in other cases is because those involve _built-in_ analysis annotations that we add for certain DML statements, which _do not_ have to be generated for every C line corresponding to the line with the DML statement.

Reply:
good! then all is fine